lib/disability_compensation/providers/document_upload/evss_supplemental_document_upload_provider.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+require 'disability_compensation/providers/document_upload/supplemental_document_upload_provider'
+
+class EVSSSupplementalDocumentUploadProvider
+  include SupplementalDocumentUploadProvider
+
+  # @param form526_submission [Form526Submission]
+  # @param file_body [String]
+  def initialize(form526_submission, file_body)
+    @form526_submission = form526_submission
+    @file_body = file_body
+  end
+
+  # Uploads to EVSS via the EVSS::DocumentsService require both the file body and an instance
+  # of EVSSClaimDocument, so we have to generate and validate that first.
+  # Note the EVSSClaimDocument class name is a misnomer; it is more accurately described as
+  # an assembly of file-related EVSS metadata, not the actual uploaded file itself
+  #
+  # @param file_name [String] The name of the file we want to appear in EVSS
+  # @param document_type [String] The VA document code, which corresponds to
+  # the type of document being uploaded ('Buddy/Lay Statement', 'Disability Benefits Questionnaire (DBQ)' etc.)
+  # These types are mapped in EVSSClaimDocument[DOCUMENT_TYPES]
+  #
+  # @return [EVSSClaimDocument]
+  def generate_upload_document(file_name, document_type)
+    EVSSClaimDocument.new(
```

Comment:
my only concern here is that this is not very "generic-sounding" for a response from the top level provider...
will the `LighthouseClaimDocument` look similar? but this has an "evss_claim_id" attribute so i would assume not?

i would also expect, since the claim id is from VBMS, it doesn't have to say "evss"; this can probably just be "`claim_id`".

can this turn into `ClaimDocument`, "evss_claim_id" turn into "claim_id" and then use as a return value for all providers? 🤔

Reply:
My bad I should have explained here what the Lighthouse implementation will look like: there already is a Lighthouse equivalent of this (poorly named) document metadata class,[LighthouseDocument]( https://github.com/department-of-veterans-affairs/vets-api/blob/88297-disability-benefits-nb-implement-supplemental-document-upload-provider/app/models/lighthouse_document.rb), which the mobile team created as part of their migration to the same endpoint. It's essentially the same with small differences but they have built their migration around it so I think we're probably stuck with holding the validation logic there unless we want to maintain something ourself.

I agree it would probably be best if there was a more generic class that covered both cases but I think the Lighthouse version has evolved and will continue to evolve. Let me know if that logic makes sense or if you think we might be able to push in another direction here